source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1632 @@
+
+          MPI_Bcast(&V[0], array_size, MPI_DOUBLE, 0, this->get_mpi_communicator());
+        }
+
+      // Get the sizes needed for a data table of the mesh velocities.
+      TableIndices<dim> size_idx;
+      for (unsigned int d=0; d<dim; ++d)
+        {
+          size_idx[d] = table_intervals[d]+1;
+        }
+
+      // Initialize a table to hold all velocity values that will be interpolated back to ASPECT.
+      Table<dim,double> velocity_table = fill_data_table(V, size_idx, nx, ny);
+
+      // As our grid_extent variable end points do not account for the change related to an origin
+      // not at 0, we adjust this here into an interpolation extent.
+      std::array<std::pair<double,double>,dim> interpolation_extent;
+      for (unsigned int i=0; i<dim; ++i)
+        {
+          interpolation_extent[i].first = grid_extent[i].first;
+          interpolation_extent[i].second = (grid_extent[i].second + grid_extent[i].first);
+        }
+
+      Functions::InterpolatedUniformGridData<dim> *velocities;
+      velocities = new Functions::InterpolatedUniformGridData<dim> (interpolation_extent,
+                                                                    table_intervals,
+                                                                    velocity_table);
+
+      VectorFunctionFromScalarFunctionObject<dim> vector_function_object(
+        [&](const Point<dim> &p) -> double
+      {
+        return velocities->value(p);
+      },
+      dim-1,
+      dim);
+
+      VectorTools::interpolate_boundary_values (mesh_deformation_dof_handler,
+                                                *boundary_ids.begin(),
+                                                vector_function_object,
+                                                mesh_velocity_constraints);
+    }
+
+
+    template <int dim>
+    std::vector<std::vector<double>>
+    FastScape<dim>::get_aspect_values() const
+    {
+
+      const types::boundary_id relevant_boundary = this->get_geometry_model().translate_symbolic_boundary_name_to_id ("top");
+      std::vector<std::vector<double>> temporary_variables(dim+2, std::vector<double>());
+
+      // Get a quadrature rule that exists only on the corners, and increase the refinement if specified.
+      const QIterated<dim-1> face_corners (QTrapez<1>(),
+                                           pow(2,additional_refinement+surface_refinement_difference));
+
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        face_corners,
+                                        update_values |
+                                        update_quadrature_points);
+
+      typename DoFHandler<dim>::active_cell_iterator
+      cell = this->get_dof_handler().begin_active(),
+      endc = this->get_dof_handler().end();
+
+      for (; cell != endc; ++cell)
```

Comment:
There is a simpler way to write this by now:
```suggestion
for (const auto &cell : this->get_dof_handler().active_cell_iterators())
```